torch/csrc/jit/api/module.h

```diff
@@ -117,25 +117,31 @@ struct TORCH_API Module : public Object {
   // register_buffer method. With this simplification, we only need to track
   // whether a slot is a parameter to be able to classify it.
   void register_buffer(const std::string& name, at::Tensor v) {
-    type()->addOrCheckAttribute(name, TensorType::get());
+    bool is_param = false;
+    bool is_buffer = true;
+    type()->addOrCheckAttribute(name, TensorType::get(), is_param, is_buffer);
```

Comment:
Hmm…having a lot of boolean defaults in a row is getting a little scary. For example, there is a bug here, because you are actually setting `is_param = false, allow_any = true, is_buffer=false`, which is not what you want. I think either we should undefault them or change it to something more declarative than several flags

Reply:
Agreed. I know you said it earlier and I pushed back, but I find myself coming around to the idea of replacing it with our enum.

Reply:
I like your earlier suggestion of utilizing our enum. Since this would be a large (but relatively simple) change, are you fine with me moving the attribute adding APIs to the new enum in a subsequent diff?

Reply:
yeah I think that makes sense!

Reply:
This is still a bug, right?

Reply:
Yea, I thought I pushed a fix. will update.